src/ploneorg/addonlisting/utils.py

```diff
@@ -35,9 +35,15 @@ def update_addon_list(context, request=None):
 
     new_addons = set(queried_addon_list) - set(present_addon_list)
 
+    if limit:
+        new_addons = new_addons[:limit]
+
     if request is not None:
         request.response.write("Start Update Add'on Listing\n")
 
+    if verbose:
```

Comment:
I am not sure if I like this verbose checks as log levels are designed for that, but also if we would set log level on start to debug or something else, that could also print statements that we do not want to be returned.